eval/src/vespa/eval/instruction/generic_rename.cpp

```diff
@@ -101,8 +101,32 @@ void my_generic_rename_op(State &state, uint64_t param_in) {
     state.pop_push(result_ref);
 }
 
+template <typename CT>
+void my_dense_rename_op(State &state, uint64_t param_in) {
+    const auto &param = unwrap_param<RenameParam>(param_in);
+    const DenseRenamePlan &dense_plan = param.dense_plan;
+    const auto &index = state.peek(0).index();
+    auto lhs_cells = state.peek(0).cells().typify<CT>();
+    size_t num_subspaces = index.size();
+    size_t num_out_cells = dense_plan.subspace_size * num_subspaces;
+    ArrayRef<CT> out_cells = state.stash.create_uninitialized_array<CT>(num_out_cells);
+    CT *dst = out_cells.begin();
+    const CT *lhs = lhs_cells.begin();
+    auto copy_cells = [&](size_t input_idx) { *dst++ = lhs[input_idx]; };
+    for (size_t i = 0; i < num_subspaces; ++i) {
+        dense_plan.execute(0, copy_cells);
+        lhs += dense_plan.subspace_size;
+    }
+    assert(lhs == lhs_cells.end());
+    assert(dst == out_cells.end());
+    state.pop_push(state.stash.create<ValueView>(param.res_type, index, TypedCells(out_cells)));
+}
+
 struct SelectGenericRenameOp {
-    template <typename CT> static auto invoke() {
+    template <typename CT> static auto invoke(const RenameParam &param) {
+        if (param.sparse_plan.can_forward_index) {
+            return my_dense_rename_op<CT>;
```

Comment:
consider using a name that reflects that this is not a simple dense rename (multi_dense for example)